File: preference/src/main/java/de/maxr1998/modernpreferences/preferences/colorpicker/renderer/ColorWheelRenderer.java
package de.maxr1998.modernpreferences.preferences.colorpicker.renderer;

import java.util.List;

import de.maxr1998.modernpreferences.preferences.colorpicker.ColorCircle;

public interface ColorWheelRenderer {
    float GAP_PERCENTAGE = 0.025f;

    void draw();

    ColorWheelRenderOption getRenderOption();

    void initWith(ColorWheelRenderOption colorWheelRenderOption);

    List<ColorCircle> getColorCircleList();
}

File: preference/src/main/java/de/maxr1998/modernpreferences/preferences/colorpicker/renderer/ColorWheelRenderer.kt
package de.maxr1998.modernpreferences.preferences.colorpicker.renderer

import de.maxr1998.modernpreferences.preferences.colorpicker.ColorCircle

interface ColorWheelRenderer {
    fun draw()
    val renderOption: ColorWheelRenderOption
    fun initWith(colorWheelRenderOption: ColorWheelRenderOption)
    fun colorCircleList(): List<ColorCircle>

    companion object {
        const val GAP_PERCENTAGE = 0.025f
    }
}